### ISSUE TITLE
How to use it?

### ISSUE THREAD

**Issue by naser** [dcouturier/LibraryOverrider#1]
Hi,
I am looking to write a library overrider to override any library by giving its .h and .so file. Is this LibraryOverrider works like the one that I would like to write or it is specialized for the opencli files? How to use it, if it supports any .h file!?
Thanks

**Comment by dcouturier** [comment]
Oh boy looking at this Java code I wrote years ago is hard lol.

So the Java code generates a wrapper for all function prototypes in the .h header file and adds some tracepoints at the beginning and end of the functions in the .c file. 
You need to dlopen this library and it will load and use the real library installed on the computer (in this case look at the __constructor__ (https://github.com/dcouturier/LibraryOverrider/blob/master/clust.c#L106)
This is where all symbols from the real library are forwarded to the ones in clust.c which adds the tracing. You should set up a meeting with me is you'd like to talk about this.

**Comment by naser** [comment]
Yes, I see that it makes an overloading library for a given header file. But how to call the java file for a given .h file? Suppose I have a new .h file, how can I use this to generate the new files (tpp and the .c file having the tracepoint codes) ?! Do you remember this?
Thanks :)

**Comment by dcouturier** [comment]
This directory is a Java eclipse project. (https://github.com/dcouturier/LibraryOverrider/blob/master/.project)

I think you just need to import it as a project and run the UI.Javai class. There will be a UI to select the .h input file. Also you may want to change the instrumentation naming which seems to be hard coded in UI.Java.. this code makes my eyes bleed... :/
